.pre-commit-config.yaml

```diff
@@ -26,10 +26,10 @@ repos:
                 files: python/.*\.(py|pyx|pxd)$
                 types: [file]
       - repo: https://github.com/pre-commit/mirrors-mypy
-        rev: 'v0.782'
+        rev: 'v0.971'
         hooks:
               - id: mypy
-                args: ["--config-file=setup.cfg", "python/cudf/cudf", "python/dask_cudf/dask_cudf", "python/custreamz/custreamz", "python/cudf_kafka/cudf_kafka"]
+                args: ["--config-file=setup.cfg", "--explicit-package-bases", "--namespace-packages", "python/cudf/cudf", "python/dask_cudf/dask_cudf", "python/custreamz/custreamz", "python/cudf_kafka/cudf_kafka"]
```

Comment:
Can you explain why `--namespace-packages --explicit-package-bases` are needed here? https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-namespace-packages

Reply:
The `test` and `benchmarks` directories don't contain `__init__.py` and so mypy complains about multiple `conftest` modules:
```
$ mypy --config-file=setup.cfg python/cudf
python/cudf/benchmarks/conftest.py: error: Duplicate module named "conftest" (also at "python/cudf/benchmarks/internal/conftest.py")
python/cudf/benchmarks/conftest.py: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#mapping-file-paths-to-modules for more info
python/cudf/benchmarks/conftest.py: note: Common resolutions include: a) using `--exclude` to avoid checking one of them, b) adding `__init__.py` somewhere, c) using `--explicit-package-bases` or adjusting MYPYPATH
Found 1 error in 1 file (errors prevented further checking)
```

Reply:
Ah yes I remember seeing this error before. My original plan was to address this by moving the tests/benchmarks outside the main repo, but that's a bigger task that requires more coordination. I'm fine with this solution as a stopgap.

Reply:
We should just `--exclude` tests and benchmarks, right? We don't need to run mypy on those.

Reply:
We ignore all the errors in the tests subdir, so I guess we could do that.

Reply:
Done in 114b4b1bdf